1.1.0 Main fails to start in IntelliJ

camel-quarkus-examples-timer-log-main fails to start in Intelij:

```
Caused by: java.lang.IllegalArgumentException: The supplied 'main-class' value of 'org.apache.camel.quarkus.main.CamelMainApplication' does not correspond to either a fully qualified class name or a matching 'name' field of one of the '@QuarkusMain' annotations
```

Stacktrace:

```
at io.quarkus.builder.Execution.run(Execution.java:116)
at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:149)
at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:250)
... 18 more
Caused by: java.lang.IllegalArgumentException: The supplied
'main-class' value of
'org.apache.camel.quarkus.main.CamelMainApplication' does not
correspond to either a fully qualified class name or a matching 'name'
field of one of the '@QuarkusMain' annotations
at io.quarkus.deployment.steps.MainClassBuildStep.mainClassBuildStep(MainClassBuildStep.java:400)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native
Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)
at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
at java.base/java.lang.Thread.run(Thread.java:832)
at org.jboss.threads.JBossThread.run(JBossThread.java:479)

2020-09-16 16:35:45,823 INFO  [io.qua.dep.dev.IsolatedDevModeMain]
(main) Attempting to start hot replacement endpoint to recover from
previous Quarkus startup failure
```

--- a/extensions-core/main/deployment/src/main/java/org/apache/camel/quarkus/main/deployment/CamelMainProcessor.java
+++ b/extensions-core/main/deployment/src/main/java/org/apache/camel/quarkus/main/deployment/CamelMainProcessor.java
@@ -28,6 +28,7 @@ import io.quarkus.deployment.annotations.Consume;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Overridable;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.AdditionalIndexedClassesBuildItem;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.QuarkusMain;
@@ -42,6 +43,7 @@ import org.apache.camel.quarkus.core.deployment.spi.CamelRuntimeTaskBuildItem;
 import org.apache.camel.quarkus.core.deployment.spi.ContainerBeansBuildItem;
 import org.apache.camel.quarkus.core.deployment.spi.RuntimeCamelContextCustomizerBuildItem;
 import org.apache.camel.quarkus.main.CamelMain;
+import org.apache.camel.quarkus.main.CamelMainApplication;
 import org.apache.camel.quarkus.main.CamelMainConfig;
 import org.apache.camel.quarkus.main.CamelMainProducers;
 import org.apache.camel.quarkus.main.CamelMainRecorder;
@@ -182,5 +184,11 @@ public class CamelMainProcessor {
                         main.getInstance(),
                         camelMainConfig.shutdown.timeout.toMillis()),
                 index.getIndex().getAnnotations(DotName.createSimple(QuarkusMain.class.getName())).isEmpty());
+    }
+
+    @BuildStep
+    AdditionalIndexedClassesBuildItem indexCamelMainApplication() {
+        // Required for launching CamelMain based applications from the IDE
+        return new AdditionalIndexedClassesBuildItem(CamelMainApplication.class.getName());
     }
 }

